### DISCUSSION TITLE
Linear to annular

### DISCUSSION THREAD

**Discussion by JeoaFesketto** [NAnand-TUD/parablade#18] [Q&A]
Hello Nitish,

First of all, I would like to thank you and your collaborators for this code which has been making my work a lot easier.

I am currently trying to initialise cfg files from the point cloud directly by setting the coordinates of the leading and trailing edge (+ stagger) control points. 

When doing this for the linear case, this would be quite straight forward:

assuming we have `le`, the point for the leading edge in cartesian coordinates from my point cloud of a linear cascade, `x, y, z = le[0], le[1], le[2]` would do.

However, I haven't been able to do so for points from an annular cascade... I am getting troubles with inferring the right value of y. For z, I have simply been taking the arc length which seems to work.

I am guessing this is to do with eq 28 of your paper, which, if I'm not mistaken, is found using the small angles approximation.
It might be important to note that my profiles vary radially along x, ie. they are on a cone rather than a cylinder. 

I've attached some screenshots to make things clearer (on the r37, the radius doesn't vary linearly along x but I am not too bothered by this, a linear variation is fine for me)

Would you be able to help me with this?
Don't hesitate if you want me to explain this further or try to provide a simple example.

Best regards, 
Jean

![Figure_1](https://user-images.githubusercontent.com/63069327/202750819-a0b6650b-e7c0-46cd-946f-b84b457b0dfd.png)
As you can see on the above figure, I wasn't able to match the trailing edge properly after using the following:
`z1 = np.linalg.norm([le[1], le[2]])`
`y1 = 2*(z1*np.arctan((z1-le[2])/le[1]))`
`z2 = np.linalg.norm([te[1], te[2]])`
`y2 = 2*(z2*np.arctan((z2-te[2])/te[1]))`

`config["x_leading"] = np.array([le[0]])`
`config["y_leading"] = np.array([y1])`
`config["z_leading"] = np.array([z1])`
`config["x_trailing"] = np.array([te[0]])`
`config["z_trailing"] = np.array([z2])`

`config["stagger"] = np.arctan((y2 - y1) / (te[0] - le[0]))`

![Figure_2](https://user-images.githubusercontent.com/63069327/202750827-41b8fc7c-2815-4af3-9323-c6c1e3ebf6dc.png)

**Comment by NAnand-TUD** [comment]
Hello! 

So we have an executable called Match Blade. If you give this code a good initial guess it will get you the correct configuration file for your cloud point. It works for 2D and 3D. You have to give the MatchBlade code a file with point cloud (INDEX X-coord y-coord z-coord). I hope this helps. Apologies for the delay.

Thanks,
Nitish

**Comment by o4fr** [comment]
Hi @JeoaFesketto,

I have experienced the same problem. Could this be solved by implementing the transformation described in the following article instead of Equation 28 in the paper? [https://www.jstor.org/stable/27034064](https://www.jstor.org/stable/27034064).

**Reply by JeoaFesketto** [reply]
Hi @o4fr, 
Could you specify which equation/paragraph in the paper you cited? It's been a while since I looked into this, I don't even understand the solution I found anymore.

**Reply by o4fr** [reply]
Hi again,

After further investigation I found a more detailed reference, see [1], which in detail explains how the 2D profile may be distorted when mapped onto a general surface of revolution. As far as I can understand, Parablade currently has two mappings implemented namely the LINEAR and ANNULAR cascade. For LINEAR cascade, there is a linear mapping and hence no distortion of the 2D profile sections. For the ANNULAR cascade mapping Equation 28 is applied and angles in 2D space is not preserved unless the surface of revolution is close to the shape of a cylinder(?). As suggested by Nitish the correct values of the config can be retrieved by applying blade matching. However, it would be useful to have implementation of the mappings commonly used within turbomachinery design e.g. (m, r*&#952;) -> (r, z, &#952;), especially when blade is designed from scratch and there is no blade to match against. I will look into implementing the mapping described in [1].

Gulich [2] also describes common mapping applied for pump design, namely the Kaplan-method. See section 7.2.2.2 in [2].

[1] Miller IV, Perry Lennox. Blade geometry description using B-splines and general surfaces of revolution. Iowa State University, 2000.

[2] Gülich, Johann Friedrich. Centrifugal pumps. Springer Berlin Heidelberg, 2008.

**Reply by NAnand-TUD** [reply]
Very interesting, let me know if you need any support to implement this in the code.
